### ISSUE TITLE
[Bug] Error on listing child items from a collection

### ISSUE THREAD

**Issue by derevandal** [netlify/netlify-cms#2414]
**Describe the bug**
When I access a collection, Netlify CMS list only one item, but, I know there are more and, When I open this only item and go back, all other items appear.

- `/.netlify/git/gitlab/repository/files/.lfsconfig/raw?ref=master&ts=1561652962650`: this api returns an error when a open one collection
- `/.netlify/git/gitlab/repository/tree?id=andrevandal%2Finstitutoalves-new&page=1&path=content%2Fstore&per_page=20&recursive=false&ref=master&ts=1561653109770`: this api returns all items when I go back

**To Reproduce**
Steps to reproduce the behavior. For example:
1. Select some Collection
2. Only one item will apper
3. Open this item
4. Go back
5. See all items

**Expected behavior**
List all child items from a collection

**Screenshots**
[![Image from Gyazo](https://i.gyazo.com/eba0955352e9d84aac4a320687cacc7c.gif)](https://gyazo.com/eba0955352e9d84aac4a320687cacc7c)
![image](https://user-images.githubusercontent.com/1340508/60283893-a29ff900-98e0-11e9-8093-205611429110.png)

**Applicable Versions:**
 - Netlify CMS version: 2.9.3
 - Git provider: [e.g. GitHub, BitBucket]
 - OS: [e.g. Windows 7]
 - Browser version [e.g. chrome 22, safari 11]

 - Node.JS version:

**CMS configuration**
```yml
backend:
  name: git-gateway
site_url: https://institutoalves.com.br
media_folder: "static/uploads"
public_folder: "/uploads"
show_preview_links: false

slug:
  encoding: "ascii"
  clean_accents: true

collections:
  - name: "talks"
    label: "Palestras"
    folder: "content/talks"
    create: true
    slug: "palestras/{{slug}}"
    fields:
      - {label: "Título", name: "title", widget: "string" }
      - {label: "Link", name: "link", widget: "string", default: "/contato" }
      - {label: "Imagem", name: "image", widget: "image" }
      - {label: "Descrição", name: "description", widget: "string" }
      - {label: "Rascunho", name: "draft", widget: "boolean", default: true}
      - {label: "Conteúdo", name: "body", widget: "markdown" }
    editor:
      preview: false
  - name: "store"
    label: "Produtos"
    folder: "content/store"
    create: true
    slug: "{{type}}/{{slug}}"
    fields:
      - {label: "Título", name: "title", widget: "string"}
      - label: "Preço"
        name: "price"
        widget: "object"
        fields:
          - {label: "De:", name: "from", widget: string }
          - {label: "Por:", name: "to", widget: string }
      - {label: "Link", name: "link", widget: "string"}
      - {label: "Imagem", name: "image", widget: "image"}
      - label: "Tipo"
        name: "type"
        widget: "select"
        options:
          - {label: "Brinquedos Pedagógicos", value: "brinquedos-pedagogicos" }
          - {label: "Livros", value: "livros" }
          - {label: "Outros", value: "outros" }
      - {label: "Rascunho", name: "draft", widget: "boolean", default: true}
      - {label: "Conteúdo", name: "body", widget: "markdown"}
    editor:
      preview: false
  - name: "categories"
    label: "Categorias"
    folder: "content/categories"
    create: true
    slug: "{{slug}}"
    fields:
      - {label: "Título", name: "title", widget: "string"}
      - {label: "Slug", name: "slug", widget: "string"}
  - name: "blog"
    label: "Publicações"
    folder: "content/posts"
    create: true
    slug: "blog/{{slug}}"
    fields:
      - {label: "Título", name: "title", widget: "string"}
      - {label: "Imagem", name: "image", widget: "image"}
      - {label: "Rascunho", name: "draft", widget: "boolean", default: true}
      - {label: "Categoria",
        name: "category",
        widget: "relation",
        collection: "categories",
        searchFields: "title",
        valueField: "slug",
        displayFields: "title"
      }
      - {label: "Conteúdo", name: "body", widget: "markdown"}
  - label: "Configurações"
    name: "configs"
    create: false
    files:
      - label: "Configurações Gerais"
        name: "settings-general"
        file: "content/settings.yml"
        fields:
          - {label: "URL Vídeo do Youtube da Página Inicial", name: "homeYoutubeURL", widget: "string"}
          - {label: "URL do WhatsApp da Página de Contato", name: "contactWhatsappUrl", widget: "string"}
      - label: "Página: Sobre Nós"
        name: "about-us"
        file: "content/pages/sobre-nos.md"
        fields:
          - {label: "Título", name: "title", widget: "string"}
          - {label: "Imagem", name: "image", widget: "image"}
          - {label: "Descrição", name: "description", widget: "string" }
          - {label: "Conteúdo", name: "body", widget: "markdown"}
    editor:
      preview: false

```

**Comment by sparkalow** [comment]
I'm having the same issue. It may be related to the number of items/pagination coming from my Gitlab backend. If I have 40 items everything seems fine.  If I have 41 items or more this bug manifests.

I have created a barebones example repo that exhibits this behavior. You can use the select item -> go back workaround outlined by @derevandal, or remove items from the collection to bring the total collection entries to less than 40 to stop the bug.

Here is the public repo for testing.
https://gitlab.com/sparkalow/netlifycms-issue

**Comment by erquhart** [comment]
Sounds like a GitLab pagination issue.

@derevandal you can ignore the `.lfsconfig` 404, it's unrelated. That's just Netilfy CMS checking to see if your repo is using Git LFS, the 404 is expected if you aren't.

@sparkalow thank you so much for the repro case!

**Comment by erquhart** [comment]
Troubleshooting this right now. I loaded the GitLab backend locally and got a completely different issue where it errors out trying to load the collection in the first place.

**Comment by erquhart** [comment]
Fixed in 2.9.5.
